numpy/tests/test_typing.py

```diff
@@ -0,0 +1,127 @@
+import importlib.util
```

Comment:
Test runner is a copy paste from https://github.com/numpy/numpy-stubs/blob/master/tests/test_stubs.py, but with some modifications:

- In the `numpy-stubs` repo the tests were run against an installed version of the stubs (because you can't point at `*-stubs` repo using `MYPYPATH`)
- Now we can use `MYPYPATH` to point at NumPy in the test env, so do that so that the tests can run in the normal way
- This does mean that mypy is type checking all of NumPy, so we will get complaints about NumPy internals that we don't want to hear
- To fix that we use a `mypy.ini` file that silences them
- By default errors from inside installed third-party libraries are silenced, so those internal type errors shouldn't bother end-users.